GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/StorageClient.UploadObject.cs

```diff
@@ -0,0 +1,197 @@
+        /// </summary>
+        /// <param name="bucket">The name of the bucket in which to store the object. Must not be null.</param>
+        /// <param name="objectName">The name of the object within the bucket. Must not be null.</param>
+        /// <param name="contentType">The content type of the object. Must not be null.</param>
+        /// <param name="source">The stream to read the data from. Must not be null.</param>
+        /// <returns>The metadata for the uploaded object.</returns>
+        public Object UploadObject(string bucket, string objectName, string contentType, Stream source)
+        {
+            return UploadObject(bucket, objectName, contentType, source, options: null, progress: null);
+        }
+
+        /// <summary>
+        /// Uploads the data for an object in storage synchronously, from a specified stream.
+        /// </summary>
+        /// <param name="bucket">The name of the bucket containing the object. Must not be null.</param>
+        /// <param name="objectName">The name of the object within the bucket. Must not be null.</param>
+        /// <param name="contentType">The content type of the object. Must not be null.</param>
+        /// <param name="source">The stream to read the data from. Must not be null.</param>
+        /// <param name="options">Additional options for the upload. May be null, in which case appropriate
+        /// defaults will be used.</param>
+        /// <param name="progress">Progress reporter for the upload. May be null.</param>
+        /// <returns>The metadata for the uploaded object.</returns>
+        public Object UploadObject(
+            string bucket,
+            string objectName,
+            string contentType,
+            Stream source,
+            UploadObjectOptions options,
+            IProgress<IUploadProgress> progress)
+        {
+            ValidateBucket(bucket);
+            objectName.CheckNotNull(nameof(objectName));
+            contentType.CheckNotNull(nameof(contentType));
+            return UploadObject(
+                new Object { Bucket = bucket, Name = objectName, ContentType = contentType },
+                source, options, progress);
+        }
+
+        /// <summary>
+        /// Uploads the data for an object in storage asynchronously, from a specified stream.
+        /// </summary>
+        /// <param name="bucket">The name of the bucket containing the object. Must not be null.</param>
+        /// <param name="objectName">The name of the object within the bucket. Must not be null.</param>
+        /// <param name="contentType">The content type of the object. Must not be null.</param>
+        /// <param name="source">The stream to read the data from. Must not be null.</param>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public Task<Object> UploadObjectAsync(string bucket, string objectName, string contentType, Stream source)
```

Comment:
What is the comment pattern for the fact that this `Task` object actually evaluates to a result, not just representing the operation?